iceoryx_examples/singleprocess/single_process.cpp

```diff
@@ -44,39 +45,57 @@ void consoleOutput(const std::string& output)
 
 void sender()
 {
-    iox::popo::Publisher publisher({"Single", "Process", "Demo"});
+    iox::popo::UntypedPublisher publisher({"Single", "Process", "Demo"});
     publisher.offer();
 
     uint64_t counter{0};
     while (keepRunning.load())
     {
-        auto sample = static_cast<TransmissionData_t*>(publisher.allocateChunk(sizeof(TransmissionData_t)));
-        sample->counter = counter++;
-        consoleOutput(std::string("Sending: " + std::to_string(sample->counter)));
-        publisher.sendChunk(sample);
+        publisher.loan(sizeof(TransmissionData_t)).and_then([&](iox::popo::Sample<void>& sample) {
+            auto rawSample = static_cast<TransmissionData_t*>(sample.get());
+            rawSample->counter = counter++;
+            consoleOutput(std::string("Sending: " + std::to_string(rawSample->counter)));
+            sample.publish();
+        });
 
         std::this_thread::sleep_for(std::chrono::milliseconds(100));
     }
 }
 
 void receiver()
 {
-    iox::popo::Subscriber subscriber({"Single", "Process", "Demo"});
+    iox::popo::UntypedSubscriber subscriber({"Single", "Process", "Demo"});
 
     uint64_t cacheQueueSize = 10;
     subscriber.subscribe(cacheQueueSize);
 
     while (keepRunning.load())
     {
-        if (iox::popo::SubscriptionState::SUBSCRIBED == subscriber.getSubscriptionState())
+        if (iox::SubscribeState::SUBSCRIBED == subscriber.getSubscriptionState())
         {
-            const void* rawSample = nullptr;
-            while (subscriber.getChunk(&rawSample))
+            bool hasMoreSamples{true};
+            const TransmissionData_t* receivedSample{nullptr};
+
+            do
             {
-                auto sample = static_cast<const TransmissionData_t*>(rawSample);
-                consoleOutput(std::string("Receiving : " + std::to_string(sample->counter)));
-                subscriber.releaseChunk(rawSample);
-            }
+                subscriber.take()
+                    .and_then([&](iox::popo::Sample<const void>& sample) {
+                        receivedSample = static_cast<const TransmissionData_t*>(sample.get());
```

Comment:
please use `auto receivedSample` and remove the line with `const TransmissionData_t* receivedSample{nullptr};` since that's a dangling pointer when used outside of this scope

Reply:
Yeah, Same as in the iceperf example! Done.

Reply:
yes, but would be better to use a typed publisher. Not sure if that should go into an separate issue to get this PR closed, though

Reply:
maybe you mixed it up with a different location. It's still not changed here :)

Reply:
Didn't commit it ;) Done now.

Reply:
maybe I'm blind, but I still can't see the new code :P

Reply:
Commit 32f3f6a ("iox-#252 Remove dangling ptr")?

Reply:
Using typed subscriber and direct access to sample!

Reply:
oh, sorry. I think I just had an old diff open